Lib/test/test_capi/test_opt.py

```diff
@@ -1767,6 +1767,20 @@ def testfunc(n):
         self.assertNotIn("_GUARD_TOS_UNICODE", uops)
         self.assertIn("_BINARY_OP_ADD_UNICODE", uops)
 
+    def test_call_type_1(self):
+        def testfunc(n):
+            for _ in range(n):
+                x = type(42)
+            return x
+
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(res, int)
```

Comment:
This really only checks that the behavior of the last loop is correct. Maybe something like this would be better, since it checks them all?

```suggestion
            x = 0
            for _ in range(n):
                x += type(42) is int
            return x

        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
        self.assertEqual(res, TIER2_THRESHOLD)
```

Reply:
ahh I see, we want to make sure that each iteration produces the expected result (sounds obvious when I write it like this :smile: )